docs/reference/mapping/types/json.asciidoc

```diff
@@ -0,0 +1,173 @@
+indexed as a string keyword, without analysis or special handling of numeric and date values.
+
+Querying the top-level `json` field searches all leaf values in the object:
+[source,js]
+--------------------------------
+POST bug_reports/_search
+{
+  "query": {
+    "term": {"labels": "urgent"}
+  }
+}
+--------------------------------
+// CONSOLE
+// TEST[continued]
+
+To query on a specific key in the JSON object, object dot notation is used:
+[source,js]
+--------------------------------
+POST bug_reports/_search
+{
+  "query": {
+    "term": {"labels.release": "v1.3.0"}
+  }
+}
+--------------------------------
+// CONSOLE
+// TEST[continued]
+
+==== Supported operations
+
+Currently, `json` fields can be used with the following query types:
+
+- term family (`term`, `terms`, `terms_set`)
+- `prefix`
+- `range`
+- match family (`match`, `match_phrase_prefix`, match_prefix`, `multi_match`)
+- `query_string` and `simple_query_string`
+- `exists`
+
+When querying, it is not possible to refer to field keys using wildcards, as in
+`{ "term": {"labels.time*": 1541457010}}`. Note that all queries, including `range`, treat the values as
+string keywords.
+
+Aggregating, highlighting, or sorting on a `json` field is not supported.
+
+Finally, because of the way leaf values are stored in the index, the null character `\0` is not allowed to
+appear in the keys of the JSON object.
+
+[[stored-fields]]
+==== Stored fields
+
+If the <<mapping-store,`store`>> option is enabled in the mappings, the entire JSON object will be stored
+in pretty-printed format. It can be retrieved through the top-level `json` field:
+
+[source,js]
+--------------------------------
+POST bug_reports/_search
+{
+  "query": { "match": { "title": "results not sorted" }},
+  "stored_fields": ["labels"]
+}
+--------------------------------
+// CONSOLE
+// TEST[continued]
+
+Field keys cannot be used to load stored content. For example, specifying
+`"stored_fields": ["labels.timestamp"]` will return an empty list.
+
+[[json-params]]
+==== Parameters for JSON fields
+
+Because of the similarities in the way values are indexed, the `json` type shares many mapping options
+with <<keyword, `keyword`>>. The following parameters are accepted:
+
+[horizontal]
+
+<<mapping-boost,`boost`>>::
+
+    Mapping field-level query time boosting. Accepts a floating point number, defaults
+    to `1.0`.
+
+`depth_limit`::
+
+    The maximum allowed depth of the JSON field, in terms of nested inner objects.
+    If a JSON field exceeds this limit, then an error will be thrown. Defaults to `20`.
+
+<<ignore-above,`ignore_above`>>::
+
+    Leaf values longer than this limit will not be indexed. Defaults to the maximum integer value
+    `2147483647` so that no values are skipped. Note that this limit applies to the leaf values
+    within the JSON field, and not the length of the entire field.
```

Comment:
Maybe we should just state that the default is to not skip values of any length rather than mentioning the explicit value here?

Reply:
Will fix, I copied this over from the `keyword` documentation but reading it over again it is quite strange!